lib/logger.js

```diff
@@ -6,449 +8,449 @@
+      }
+
+      if (typeof opts.meta === 'object') {
+        opts.meta = {...message.meta, ...opts.meta}
+        if (opts.index_meta || (opts.index_meta === undefined && this._index_meta)) {
+          message.meta = opts.meta
         } else {
-            if (typeof opts !== 'object') {
-                this._err = true;
-                debug('Can only pass a String or JSON object as additional parameter');
-            }
-
-            message.level = opts.level || message.level;
-            message.app = opts.app || message.app;
-            message.env = opts.env || message.env;
-
-            if (opts.timestamp && isValidTimestamp(opts.timestamp)) {
-                message.timestamp = opts.timestamp;
-            }
-
-            if (opts.context && !opts.meta) {
-                opts.meta = opts.context;
-            }
-
-            if (typeof opts.meta === 'object') {
-                opts.meta = Object.assign({}, message.meta, opts.meta);
-                if (opts.index_meta || (opts.index_meta === undefined && this._index_meta)) {
-                    message.meta = opts.meta;
-                } else {
-                    message.meta = stringify(opts.meta);
-                }
-            }
-
-            if (opts.logSourceCRN) {
-                message.logSourceCRN = opts.logSourceCRN;
-            }
-
-            if (opts.saveServiceCopy) {
-                message.saveServiceCopy = opts.saveServiceCopy;
-            }
-
-            if (opts.appOverride) {
-                message.appOverride = opts.appOverride;
-            }
-
-            if (this._shimProperties) {
-                this._shimProperties.forEach((prop) => {
-                    if (opts.hasOwnProperty(prop)) {
-                        message[prop] = opts[prop];
-                    }
-                });
-            }
+          message.meta = stringify(opts.meta)
         }
-    } else if (!this._index_meta) {
-        message.meta = stringify(message.meta);
-    }
-
-    if (this._err) {
-        return this._err;
-    }
+      }
 
-    this._bufferLog(message, callback);
-};
-
-Logger.prototype._bufferLog = function(message, callback) {
-    if (!message || !message.line) {
-        debug('Ignoring empty message');
-        return;
-    }
-
-    if (!callback || typeof callback !== 'function') {
-        callback = (err) => { debug(err); };
-    }
+      if (opts.logSourceCRN) {
+        message.logSourceCRN = opts.logSourceCRN
+      }
 
-    if (this._max_length && message.line.length > configs.MAX_LINE_LENGTH) {
-        message.line = message.line.substring(0, configs.MAX_LINE_LENGTH) + ' (cut off, too long...)';
-        debug('Line was longer than ' + configs.MAX_LINE_LENGTH + ' chars and was truncated.');
-    }
+      if (opts.saveServiceCopy) {
+        message.saveServiceCopy = opts.saveServiceCopy
+      }
 
-    this._bufByteLength += sizeof(message);
+      if (opts.appOverride) {
+        message.appOverride = opts.appOverride
```

Comment:
I'm wondering how we can hide `IBM`-specific `logSourceCRN`, `saveServiceCopy`, and `appOverride` - probably, not a part of this PR but this library is public and for general audience not just for IBM; so, it may cause some confusion for Open-Source contributors.

Reply:
Perhaps IBM should have a fork that's just for them.  We could choose to merge certain changes or not, and they could maintain their own list of private commits.

Reply:
We may need to discuss it later - no harm for now tho